From 0a3d16703d6df888dcbd7d59d5e183dd8f5d9ad4 Mon Sep 17 00:00:00 2001
From: Frank Jeschke <fjeschke@synapticon.com>
Date: Fri, 17 Jan 2020 15:55:29 +0100
Subject: [PATCH 1/9] Add individual request of slave operational state

---
 include/ecrt.h | 12 ++++++++++++
 lib/master.c   | 15 +++++++++++++++
 2 files changed, 27 insertions(+)

diff --git a/include/ecrt.h b/include/ecrt.h
index a0b48ae4..68d0b378 100644
--- a/include/ecrt.h
+++ b/include/ecrt.h
@@ -1167,6 +1167,18 @@ int ecrt_master_link_state(
                                        */
         );
 
+/** Set individual slave to state
+ *
+ * Request slave state during operation.
+ *
+ * \return Zero on success, otherwise negative error code
+ */
+int ecrt_master_slave_link_state_request(
+        const ec_master_t *master, /**< EtherCAT master */
+         uint16_t slave_position,  /**< slave position of slave to request */
+         uint8_t state             /**< State to request slave to go into */
+        );
+
 /** Sets the application time.
  *
  * The master has to know the application's time when operating slaves with
diff --git a/lib/master.c b/lib/master.c
index 29dcbee8..90da0ccc 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -856,6 +856,21 @@ int ecrt_master_link_state(const ec_master_t *master, unsigned int dev_idx,
     return 0;
 }
 
+int ecrt_master_slave_link_state_request(const ec_master_t *master, uint16_t slave_position, uint8_t state)
+{
+    ec_ioctl_slave_state_t data;
+    data.slave_position = slave_position;
+    data.al_state = state;
+
+    int ret = ioctl(master->fd, EC_IOCTL_SLAVE_STATE, &data);
+    if (EC_IOCTL_IS_ERROR(ret)) {
+        fprintf(stderr, "Failed to request slave state: %d", state);
+        return -EC_IOCTL_ERRNO(ret);
+    }
+
+    return 0;
+}
+
 /****************************************************************************/
 
 void ecrt_master_application_time(ec_master_t *master, uint64_t app_time)
-- 
2.17.1

