From 3b979a70e12a03ec184466d9512e41c6aa7f547d Mon Sep 17 00:00:00 2001
From: Frank Jeschke <fjeschke@synapticon.com>
Date: Fri, 17 Jan 2020 16:01:38 +0100
Subject: [PATCH 3/9] Add sdo info access to user api for ethercat library

---
 include/ecrt.h | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/include/ecrt.h b/include/ecrt.h
index 81d496aa..9025716e 100644
--- a/include/ecrt.h
+++ b/include/ecrt.h
@@ -233,6 +233,10 @@
  */
 #define EC_COE_EMERGENCY_MSG_SIZE 8
 
+/** Size of SDO access rights field
+ */
+#define EC_SDO_ENTRY_ACCESS_COUNTER   3
+
 /******************************************************************************
  * Data types
  *****************************************************************************/
@@ -961,6 +965,35 @@ int ecrt_master_sdo_upload_complete(
         uint32_t *abort_code /**< Abort code of the SDO upload. */
         );
 
+/** Executes a SDO Info request
+ *
+ * With this request the entries of the slaves object dicionty can be read.
+ *
+ * \retval  0 Success
+ * \retval <0 On Error
+ */
+int ecrt_sdo_info_get(
+        ec_master_t *master,     /**< EtherCAT master */
+        uint16_t slave_position, /**< Slave position */
+        uint16_t sdo_position,   /**< SDO position within the dictionary */
+        ec_sdo_info_t *sdo       /**< Pointer to output structure */
+        );
+
+/** Executes a SDO Info Entry request
+ *
+ * With this request a entry of the slaves object dicionty is requested.
+ *
+ * \retval  0 Success
+ * \retval <0 On Error
+ */
+int ecrt_sdo_get_info_entry(
+        ec_master_t *master,        /**< EtherCAT master */
+        uint16_t slave_position,    /**< Slave position */
+        uint16_t index,             /**< Index of the element to request */
+        uint8_t subindex,           /**< Subindex of the element to request */
+        ec_sdo_info_entry_t *entry  /**< Pointer to output structure */
+        );
+
 /** Executes an SoE write request.
  *
  * Starts writing an IDN and blocks until the request was processed, or an
-- 
2.17.1

