From c7ff6e7877143db928b80c40e72ad4ae2627d0d6 Mon Sep 17 00:00:00 2001
From: Alexey Zakharov <alexey@vectioneer.com>
Date: Thu, 5 Dec 2019 10:42:28 +0100
Subject: [PATCH 5/8] increased datagram timeout, which is necessary for some
 devices

---
 master/globals.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/master/globals.h b/master/globals.h
index 0334179..02d357c 100755
--- a/master/globals.h
+++ b/master/globals.h
@@ -44,7 +44,7 @@
  *****************************************************************************/
 
 /** Datagram timeout in microseconds. */
-#define EC_IO_TIMEOUT 500
+#define EC_IO_TIMEOUT 1000
 
 /** SDO injection timeout in microseconds. */
 #define EC_SDO_INJECTION_TIMEOUT 10000
-- 
2.7.4

