From 3ddd9ff7c29791aea8102e88445bada5ffd12d7b Mon Sep 17 00:00:00 2001
From: Nikola Zivkovic <zivke85@gmail.com>
Date: Tue, 11 Feb 2020 15:13:41 +0100
Subject: [PATCH 6/9] SII file write support

---
 include/ecrt.h | 13 +++++++++++++
 lib/master.c   | 36 ++++++++++++++++++++++++++++++++++++
 2 files changed, 49 insertions(+)

diff --git a/include/ecrt.h b/include/ecrt.h
index ed1eeaed..42bb3a42 100644
--- a/include/ecrt.h
+++ b/include/ecrt.h
@@ -1406,6 +1406,19 @@ void ecrt_master_reset(
         ec_master_t *master /**< EtherCAT master. */
         );
 
+/**
+ * Write SII data to a slave
+ *
+ * IMPORTANT: Make sure to verify the SII data first:
+ * https://github.com/synapticon/Etherlab_EtherCAT_Master/blob/develop/tool/CommandSiiWrite.cpp#L175
+ */
+int ecrt_master_write_sii(
+        ec_master_t *master,
+        uint16_t position,
+        const uint8_t *content,
+        size_t size
+        );
+
 /******************************************************************************
  * Slave configuration methods
  *****************************************************************************/
diff --git a/lib/master.c b/lib/master.c
index 1ece010d..58b5f4db 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -1078,3 +1078,39 @@ void ecrt_master_reset(ec_master_t *master)
 }
 
 /****************************************************************************/
+
+int ecrt_master_write_sii(ec_master_t *master, uint16_t position,
+                          const uint8_t *content, size_t size)
+{
+  ec_ioctl_slave_sii_t data;
+
+  data.slave_position = position;
+  data.offset = 0;
+
+  if (!size || size % 2) {
+    fprintf(
+        stderr,
+        "Failed to write SII: Invalid data size (%ld) - must be non-zero and even.\n",
+        size);
+    return -1;
+  }
+
+  data.nwords = size / 2;
+
+  // allocate buffer and read file into buffer
+  data.words = malloc(size * sizeof(uint8_t));
+  memcpy((uint8_t*) data.words, content, size);
+
+  int ret;
+
+  ret = ioctl(master->fd, EC_IOCTL_SLAVE_SII_WRITE, &data);
+  if (EC_IOCTL_IS_ERROR(ret)) {
+    fprintf(stderr, "Failed to write SII: %s\n", strerror(EC_IOCTL_ERRNO(ret)));
+  }
+
+  free(data.words);
+
+  return ret;
+}
+
+/****************************************************************************/
\ No newline at end of file
-- 
2.17.1

