From 49ca844afafa8e5617d16324b9e4d481cc43aa04 Mon Sep 17 00:00:00 2001
From: Nikola Zivkovic <zivke85@gmail.com>
Date: Tue, 29 Nov 2022 17:20:50 +0100
Subject: [PATCH 1/2] Increase the SII write delay from 5ms to 10ms

---
 master/fsm_sii.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/master/fsm_sii.c b/master/fsm_sii.c
index 32fa882f..1a65294e 100644
--- a/master/fsm_sii.c
+++ b/master/fsm_sii.c
@@ -59,7 +59,7 @@
 
 /** Time before evaluating answer at writing [ms].
  */
-#define SII_INHIBIT 5
+#define SII_INHIBIT 10
 
 //#define SII_DEBUG
 
-- 
2.34.1

